packages/bff/src/services/purposeService.ts

```diff
@@ -0,0 +1,57 @@
+import { AppContext, WithLogger } from "pagopa-interop-commons";
+import {
+  PurposeId,
+  PurposeVersionId,
+  unsafeBrandId,
+} from "pagopa-interop-models";
+import { ApiUpdateReversePurposePayload } from "../model/types.js";
+import { PagoPaClients, Headers } from "../providers/clientProvider.js";
+import { purposeNotFound } from "../model/domain/errors.js";
+
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+export function purposeServiceBuilder(
+  purposeClient: PagoPaClients["purposeProcessClient"]
+) {
+  return {
+    async reversePurposeUpdate(
+      id: PurposeId,
+      updateSeed: ApiUpdateReversePurposePayload,
+      { logger }: WithLogger<AppContext>,
+      requestHeaders: Headers
+    ): Promise<{ purposeId: PurposeId; versionId: PurposeVersionId }> {
+      logger.info(`Updating reverse purpose ${id}`);
+      const updatedPurpose = await purposeClient.updateReversePurpose(
+        updateSeed,
+        {
+          headers: { ...requestHeaders },
+          withCredentials: true,
+          params: {
+            id,
+          },
+        }
+      );
+
+      // eslint-disable-next-line @typescript-eslint/array-type
+      type VersionState = (typeof updatedPurpose.versions)[number]["state"];
```

Comment:
Could be defined in `packages/bff/src/model/types.ts` so that we can also disable this eslint rule for the entire file (we will use it array-type a lot I suppose)